jetty-deploy/src/main/java/org/eclipse/jetty/deploy/bindings/StandardDeployer.java

```diff
@@ -40,6 +40,6 @@ public void processBinding(Node node, App app) throws Exception
         {
             throw new NullPointerException("No Handler created for App: " + app);
         }
-        app.getDeploymentManager().getContexts().addHandler(handler);
+        app.getDeploymentManager().getContexts().deployHandler(handler);
     }
```

Comment:
I think we need to add a comment somewhere that previously when StandardDeployer.processBinding() returned, the caller could be sure that the handler had been added to the contexts and would be started (if the ContextHandlerCollection was started). Now, however, the call can return and the handler may not have yet been added to the collection nor started.

Reply:
Note that the context can still be started prior to this operation completing as the server is set prior the the serial execution.

Reply:
Now I have the callbacks, I have added a blocking callback here, so the original semantics are now maintained, but without a global lock that can risk deadlock.